libraries/rush-lib/src/logic/operations/ShellOperationRunnerPlugin.ts

```diff
@@ -130,12 +139,18 @@ export function formatCommand(rawCommand: string, customParameterValues: Readonl
   }
 }
 
-export function getDisplayName(phase: IPhase, project: RushConfigurationProject): string {
+export function getDisplayName(
+  phase: IPhase,
+  project: RushConfigurationProject,
+  operation: Operation
+): string {
   if (phase.isSynthetic) {
     // Because this is a synthetic phase, just use the project name because there aren't any other phases
     return project.packageName;
   } else {
     const phaseNameWithoutPrefix: string = phase.name.slice(RushConstants.phaseNamePrefix.length);
-    return `${project.packageName} (${phaseNameWithoutPrefix})`;
+    return `${project.packageName} (${phaseNameWithoutPrefix})${
+      operation.shard ? ` - shard ${operation.shard.current}` : ''
```

Comment:
The `ShardedPhaseOperationPlugin` should just append the shard suffix to the operation name when creating the runner, rather than modifying this function.

Reply:
Done